core/client/src/cht.rs

```diff
@@ -270,13 +279,14 @@ pub fn block_to_cht_number<N: SimpleArithmetic>(cht_size: u64, block_num: N) ->
 	if block_num == N::zero() {
 		None
 	} else {
-		Some((block_num - N::one()) / As::sa(cht_size))
+		Some((block_num - N::one()) / cht_size.saturated_into())
 	}
 }
 
 /// Convert header number into CHT key.
-pub fn encode_cht_key<N: As<u64>>(number: N) -> Vec<u8> {
-	let number: u64 = number.as_();
+pub fn encode_cht_key<N: UniqueSaturatedInto<u64>>(number: N) -> Vec<u8> {
+	// why not just use Encode?
```

Comment:
We should add an issue for this.

Reply:
I have already fixed that in [`remove_as_in_cht`](https://github.com/paritytech/substrate/tree/remove_as_in_cht) branch - hope to open PR later today. Probably not worth new issue in this case?

Reply:
Yeah :)